include/string_utils.h

```diff
@@ -139,6 +140,53 @@ void reset_str(T *str) noexcept
 	terminate_str_at(str, 0);
 }
 
+// Is it an ASCII control character?
+constexpr bool is_control_ascii(const char c)
+{
+	// clang-format off
+	// ASCII control characters span the bottom 5 bits plus the DEL character.
+	// Ref:https://en.wikipedia.org/wiki/ASCII#Control_characters
+	[[maybe_unused]] constexpr char controls_first = 0b0'0000;
```

Comment:
I don't like the formatting of the 3 `constexpr` values. Would it be feasible to move `[[maybe_unused]]` to a separate line?

(but this comment is purely optional)

Reply:
Same :-) Yes, making that change :+1: